src/main/java/io/mantisrx/api/tunnel/CrossRegionHandler.java

```diff
@@ -164,10 +165,19 @@ private void handleHead(ChannelHandlerContext ctx, FullHttpRequest request) {
                 .addListener(__ -> ctx.close());
     }
 
+    private List<String> parseRegionsInUri(String uri) {
+        final String regionString = getRegion(uri);
+        if (isAllRegion(regionString)) {
+            return getTunnelRegions();
+        } else if (regionString.contains(",")) {
+            return parseRegionCsv(regionString);
+        } else {
+            return Collections.singletonList(regionString);
+        }
+    }
```

Comment:
Can you add a couple of unit tests to codify this behavior?

Reply:
added